tg/controllers/__init__.py

```diff
@@ -25,20 +21,54 @@ class Controller:
     # View is terminal vindow
     """
 
-    tg: Telegram
-
     def __init__(self, model: Model, view: View, tg: Telegram) -> None:
         self.model = model
         self.view = view
-        self.tg = tg
         self.lock = threading.Lock()
+        self.tg = tg
+        self.handlers = {
+            "updateNewMessage": self.update_new_msg,
+            "updateChatLastMessage": self.update_chat_last_msg,
+            "updateMessageSendSucceeded": self.update_msg_send_succeeded,
+            "updateFile": self.update_file,
+        }
 
     def run(self) -> None:
         try:
             self.handle_chats()
         except Exception:
             log.exception("Error happened in main loop")
 
+    def download_current_file(self):
+        msg = MsgProxy(self.model.current_msg())
+        log.debug("Downloading msg: %s", msg.msg)
+        file_id = msg.file_id
+        if file_id:
+            log.info("Downloading file: file_id=%s", file_id)
+            self.model.downloads[file_id] = (msg["chat_id"], msg["id"])
+            self.tg.download_file(file_id=file_id)
+            log.info("Downloaded: file_id=%s", file_id)
+
+    def open_current_msg(self):
+        msg = MsgProxy(self.model.current_msg())
+        log.info("Open msg: %s", msg.msg)
+        if msg.is_text:
+            text = msg["content"]["text"]["text"]
+            with NamedTemporaryFile("w") as f:
+                f.write(text)
+                f.flush()
+                with suspend(self.view) as s:
+                    s.call(["less", f.name])
+            return
+
+        path = msg.local_path
+        if path:
+            # handle with mimetype and mailcap
+            # if multiple entries in mailcap, open fzf to choose
+            with suspend(self.view) as s:
+                log.info("Opening file: %s", path)
+                s.call(["open", path])
```

Comment:
this works only for mac os, we need to use `xdg-open` when on linux 
https://stackoverflow.com/a/264401/5187245